### PR TITLE
Add readthedocs.io style documentation

### PR DESCRIPTION
**Pull request by Darhazer**
I'm trying to build a read-the-docs version similar to Rubocop, using Rubocop's own rake tasks for this.
Here are the files for the documentation. I've already setup a readthedocs project, but there's no documentation for it to fetch.
Few notes:
* ~~I have some issues with Yard not wanting to pick up the docs for all departments when I filter the departments~~
* ~~We have conflict with the Rails departments and I didn't figure out yet how to generate documentation only for RSpec/Rails~~
* ~~Once the issues above are resolved, the rake tasks can be pushed as wel and publishing documentation - automatedl~~
* The PR is published at http://rubocop-rspec.readthedocs.io/en/read-the-docs/

-----------------

Before submitting the PR make sure the following are checked:

* [ ] Feature branch is up-to-date with `master` (if not - rebase it).
* [ ] Squashed related commits together.
* [ ] Added tests.
* [ ] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes.
* [ ] The build (`bundle exec rake`) is passing.

If you have created a new cop:
* [ ] Added the new cop to `config/default.yml`.
* [ ] The cop includes examples of good and bad code.
* [ ] You have tests for both code that should be reported and code that is good.

### CODE REVIEW

**Comment by bquorning** [comment]
Could you please split into a commit with your code, and a commit with the generated files?

**Comment by Darhazer** [comment]
Done. The code of the rake tasks have the minimum required modifications over the Rubocop's one so it prints only the RSpec cops

File: mkdocs.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+site_name: "RSpec-specific analysis for your projects, as an extension to RuboCop."
+repo_url: https://github.com/backus/rubocop-rspec
+docs_dir: manual
+pages:
+- Home: index.md
+- Installation: installation.md
+- Usage: usage.md
+- Cops: cops.md
+- Cops Documentation:
```
Comment (Darhazer) [inline mkdocs.yml]: oops this should be updated as well, and perhaps extracted into it's own commit

**Comment by Darhazer** [comment]
Rebased and updated

**Comment by bquorning** [comment]
I am not 100% sure about how readthedocs.io works. Do they fetch the `manual/` folder from our repository when we notify them (webhook), or will they fetch all source and run the rake task themselves?

If we need to keep the manual up to date, perhaps the rake task should be part of our release process which I imagine would need to be formalized :-)

**Comment by Darhazer** [comment]
It works with a web hook. We can add the task to the default rake build and we can keep the readthedocs synced with master (I think RuboCop does exactly that) and not with the latest release

**Comment by bquorning** [comment]
Yeah, add to the default rake task and perhaps update the Readme and the pull request template. Is anything else needed?

**Comment by Darhazer** [comment]
@bquorning added to the default task and to the contributing noets.
We have to add a hook to https://readthedocs.org/api/v2/webhook/rubocop-rspec/28346/ ([set up documentation](https://docs.readthedocs.io/en/latest/webhooks.html))

**Comment by bquorning** [comment]
I have set up the webhook – please let me know if it works or not.

File: .circleci/config.yml [hunk 2]
```diff
@@ -13,13 +13,13 @@ rubocop: &rubocop
     - run: rake internal_investigation
 
 jobs:
-  confirm_config:
+  confirm_config_and_documentation:
```
Comment (bquorning) [inline .circleci/config.yml]: ❤️

**Comment by Darhazer** [comment]
webhook works :beers:

File: .circleci/config.yml [hunk 3]
```diff
@@ -111,28 +111,28 @@ workflows:
   version: 2
   build:
     jobs:
-      - confirm_config
+      - confirm_config_and_documentation
 
-      # Use `requires: [confirm_config]` to trick Circle CI into starting the slow jruby job early.
+      # Use `requires: [confirm_config_and_documentation]` to trick Circle CI into starting the slow jruby job early.
```
Comment (bquorning) [inline .circleci/config.yml]: I noticed how this build can be incredibly slow now, e.g. https://circleci.com/gh/rubocop-rspec/rubocop-rspec/1982

Perhaps we should use `required: [code-climate]` instead for this “trick”?

Comment (bquorning) [reply .circleci/config.yml]: Ah, I found the issue. See comment on the rake task.

File: tasks/cops_documentation.rake [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+      .join("\n")
+  end
+
+  def assert_manual_synchronized
+    # Do not print diff and yield whether exit code was zero
+    sh('git diff --quiet manual') do |outcome, _|
+      return if outcome
+
+      # Output diff before raising error
+      sh('git diff manual')
```
Comment (bquorning) [inline tasks/cops_documentation.rake]: The default pager for `git diff` is `less`. And `less` doesn’t quit, which then causes the CI to hang.

E.g. `sh('PAGER=cat git diff manual')` should fix it.

Comment (Darhazer) [reply tasks/cops_documentation.rake]: I think we don't need this assert, since we have such check in `confirm_documentation` as well

Comment (bquorning) [reply tasks/cops_documentation.rake]: True, that is exactly what `rake confirm_documentation` does.

File: .circleci/config.yml (hunk 3 above)
Comment (bquorning) [reply .circleci/config.yml]: > Perhaps we should use required: [code-climate] instead for this “trick”?

Just to be clear: Please disregard that suggestion now. :-)

**Comment by bquorning** [comment]
I have removed `ci/circleci: confirm_config` as a required status on the repository. After merging, I will enable `ci/circleci: confirm_config_and_documentation` as a new required status.
